Mathlib/Data/List/Commutativity.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2025 Shimanonogov Igor. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shimanonogov Igor
+-/
+import Mathlib.Data.List.Defs
+import Mathlib.Data.List.RepeatSelf
+
+/-!
+# commutativity
+
+Results commencted with commutativity of List with respect to append.
+
+## Main results
+
+- `comm_iff_common_root`: Lyndon-Schutzenberger theorem.
```

Comment:
Can you also explain what the Lyndon-Schutzenberg theorem states? Most people here don't know what this is, and I find at least two different statements when googling.